### PR TITLE
fixPoint document added

### PR DESCRIPTION
**Pull request by jijingg**
fixPoint will avaliable on spinalHDL-1.3.7,  PR #232 already merged

### CODE REVIEW

File: source/SpinalHDL/Data types/Int.rst [hunk 1]
```diff
@@ -416,3 +440,143 @@ Misc
...
+ roundUp       Both        simple for HW                w(x)-n+1 bits         w(x)-n bits
+ roundDown     Both                                     w(x)-n+1 bits         w(x)-n bits
+ roundToInf    SInt        most Common                  w(x)-n+1 bits         w(x)-n bits
+ roundToZero   SInt        equal to roundDown in UInt   w(x)-n+1 bits         w(x)-n bits
+ round         Both        SpinalHDL chose roundToInf   w(x)-n+1 bits         w(x)-n bits
+============= =========== ============================ ===================== ====================
+
+although `roundToInf` is very common.
+but `roundUp` with lowerest cost and good timing, almost no performance lost.   
+so `roundUp` is very recommended in your work.
```
Comment (wifasoi) [inline source/SpinalHDL/Data types/Int.rst]: Ok this tre sentence are disconnected from each other, but connected by a `but` and `so`. maybe it should be:
```
.. note::
   | Although `roundToInf` is very common rounding strategy, `roundUp` has the lower logic cost and almost no performance hit.
   | So using `roundUp` is recommended.
```

File: source/SpinalHDL/Data types/Int.rst [hunk 2]
```diff
@@ -416,3 +440,143 @@ Misc
...
+
+===================== ======= =================== ========================================================= =====================
+ Programing-language   api     RoundType(spinal)   Example                                                   comments
+===================== ======= =================== ========================================================= =====================
+ Matlab                round   ROUNDTOINF          round(1.5)=2,round(2.5)=3;round(-1.5)=-2,round(-2.5)=-3   round to ±Infinity
+ python2               round   ROUNDTOINF          round(1.5)=2,round(2.5)=3;round(-1.5)=-2,round(-2.5)=-3   round to ±Infinity
+ python3               round   ROUNDTOEVEN         round(1.5)=round(2.5)=2;  round(-1.5)=round(-2.5)=-2      close to Even
+ Scala.math            round   ROUNDTOUP           round(1.5)=2,round(2.5)=3;round(-1.5)=-1,round(-2.5)=-2   always to +Infinity
+ SpinalHDL             round   ROUNDTOINF          round(1.5)=2,round(2.5)=3;round(-1.5)=-2,round(-2.5)=-3   round to ±Infinity
+===================== ======= =================== ========================================================= =====================
```
[the remark this answers is not in the document]
Comment (jijingg) [reply source/SpinalHDL/Data types/Int.rst]: i removed the colunm of "api", i hope the example is better keeped, it's more instructive.

File: source/SpinalHDL/Data types/Int.rst [hunk 3]
```diff
@@ -416,3 +440,143 @@ Misc
...
+   val B  = A.roundToInf(6 bits)
+   val B  = A.roundToZero(6 bits)
+   val B  = A.round(6 bits)             //spinalHDL use roundToInf as default round
+
+   val B0 = roundToInf(6 bits, align=true)          ---+
+                                                       |--> equal
+   val B1 = roundToInf(6 bits, align=false).sat(1)  ---+
+
+only `floor` and `floorToZero` without align option, they do not need carry bit.
+other round operation need carry bit.
```
[the remark this answers is not in the document]
Comment (jijingg) [reply source/SpinalHDL/Data types/Int.rst]: right

File: source/SpinalHDL/Data types/Int.rst [hunk 4]
```diff
@@ -416,3 +440,143 @@ Misc
 
    // Absolute value
    mySInt_abs := mySInt.abs
+
+
+fixPoint operation
+^^^^^^^^^^^^^^^^^^
+For fixed-point, we can divide it into two parts.
+ - LowerBit Operation(round methods)
+ - HighBit Operation(saturation operations)
```
[the remark this answers is not in the document]
Comment (jijingg) [reply source/SpinalHDL/Data types/Int.rst]: Yes

**Comment by jijingg** [comment]
thanks for your detailed review @wifasoi 
I revised the doc according your suggestion.

**Comment by wifasoi** [comment]
Sorry for the late reply.
Two option:
- You can rebase the pr and change target to dev (and i will merge it)
- We can wait after the 1.3.7 release.
This is why this pr is not merged yet.

Either way this will be merged :)

**Comment by Dolu1990** [comment]
My bad, forgot about it :/

**Comment by Dolu1990** [comment]
hmmm Ahh yes, so i should probably release 1.3.7 soon, there is already quite some fixes in it :)

**Pull request merged**